src/Plugin.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Varnish\Plugin.
+ */
+
+namespace Netzstrategen\Varnish;
+
+/**
+ * Main front-end functionality.
+ */
+class Plugin {
+
+  /**
+   * Prefix for naming.
+   *
+   * @var string
+   */
+  const PREFIX = 'varnish';
+
+  /**
+   * Gettext localization domain.
+   *
+   * @var string
+   */
+  const L10N = self::PREFIX;
+
+  /**
+   * @var string
+   */
+  private static $baseUrl;
+
+  /**
+   * Loads the plugin textdomain.
+   */
+  public static function loadTextdomain() {
+    load_plugin_textdomain(static::L10N, FALSE, static::L10N . '/languages/');
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+    add_action('save_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('deleted_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('trashed_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('edit_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('delete_attachment', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('switch_theme', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('wp_update_attachment_metadata', __CLASS__ . '::purgeAttachmentMeta', 50, 2);
+
+    add_action('shutdown', __CLASS__ . '::executePurge');
+  }
+
+  /**
+   * Registers the given post, associated terms, authors, and feeds for purging.
+   *
+   * @param int $postId
+   *   The ID of the post to purge.
+   */
+  public static function purgePost($postId) {
+    // Skip post revisions.
+    if (get_permalink($postId) === FALSE) {
```

Comment:
IMO it makes sense to store the permalink in a variable to avoid multiple function calls.